index.js

```diff
@@ -101,17 +99,16 @@ UglifyWriter.prototype.build = function() {
     });
   });
 
-  return queue(worker, pendingWork, writer.concurrency)
-    .then((/* results */) => {
-      // files are finished processing, shut down the workers
-      writer.pool.terminate();
-      return writer.outputPath;
-    })
-    .catch(e => {
-      // make sure to shut down the workers on error
-      writer.pool.terminate();
-      throw e;
-    });
+  try {
+    await queue(worker, pendingWork, writer.concurrency);
+    // files are finished processing, shut down the workers
+    writer.pool.terminate();
+    return writer.outputPath;
+  } catch (e) {
+    // make sure to shut down the workers on error
+    writer.pool.terminate();
+    throw e;
+  }
```

Comment:
Let's leverage `try`/`finally` here instead of handling the termination in both branches:

```suggestion
  try {
    await queue(worker, pendingWork, writer.concurrency);
    
    return writer.outputPath;
  } finally {
    // files are finished processing, shut down the workers
    writer.pool.terminate();
  }
```

Reply:
[Updated](https://github.com/ember-cli/broccoli-uglify-sourcemap/pull/237/commits/32a6426a10141648b4e5a2df008f210446435b55) @rwjblue  👍🏽